### ISSUE TITLE
Key generation does not work because of pinentry problem

### ISSUE THREAD

**Issue by DemiMarie** [HW42/qubes-app-linux-split-gpg2#13]
If I try to generate a key, I run into a problem due to pinentry: pinentry tries to access the controlling terminal, but there is no such terminal, so it fails with a confusing error.

The simplest solution is to replace the pinentry program with a trivial script that always returns `OK` for every command.  A better solution is to always set the pinentry mode to loopback, and respond to every passphrase request with an empty response.

**Comment by HW42** [comment]
> A better solution is to always set the pinentry mode to loopback, and respond to every passphrase request with an empty response.

No. Pinentry should be supported. This should work, I will check. What pinentry are you using?

**Comment by DemiMarie** [comment]
> > A better solution is to always set the pinentry mode to loopback, and respond to every passphrase request with an empty response.
> 
> No. Pinentry should be supported. This should work, I will check. What pinentry are you using?

`pinentry-curses`, which obviously does not work, but with an error message (“Inappropriate IOCTL for device”) that is not helpful.

**Comment by marmarek** [comment]
I wonder what would happen if you try to make gpg use `pinentry-curses` from a GUI application. In other words - sounds more like a generic miss-configuration of gpg and a bit unclear error reporting - that is only easier to hit with split-gpg2, than an actual split-gpg2 issue. Close?
